azurerm/resource_arm_network_security_group.go

```diff
@@ -216,8 +219,23 @@ func resourceArmNetworkSecurityGroupDelete(d *schema.ResourceData, meta interfac
 	return err
 }
 
-func flattenNetworkSecurityRules(rules *[]network.SecurityRule) []interface{} {
-	result := make([]interface{}, 0)
+func resourceArmNetworkSecurityGroupRuleHash(v interface{}) int {
+	var buf bytes.Buffer
+	m := v.(map[string]interface{})
+	buf.WriteString(fmt.Sprintf("%s-", m["protocol"].(string)))
+	buf.WriteString(fmt.Sprintf("%s-", m["source_port_range"].(string)))
+	buf.WriteString(fmt.Sprintf("%s-", m["destination_port_range"].(string)))
+	buf.WriteString(fmt.Sprintf("%s-", m["source_address_prefix"].(string)))
+	buf.WriteString(fmt.Sprintf("%s-", m["destination_address_prefix"].(string)))
+	buf.WriteString(fmt.Sprintf("%s-", m["access"].(string)))
+	buf.WriteString(fmt.Sprintf("%d-", m["priority"].(int)))
+	buf.WriteString(fmt.Sprintf("%s-", m["direction"].(string)))
```

Comment:
since all items are contained within the Set's hash function, we actually don't need to specify this function at all - so can we remove this method? the default set hash function will include all values in the set

Reply:
Agreed. removed this function.